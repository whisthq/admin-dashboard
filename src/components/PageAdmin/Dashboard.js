import React, { Component } from 'react'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import { connect } from 'react-redux'
import 'react-tabs/style/react-tabs.css'

import '../../static/App.css'
import {
    fetchUserActivity,
    fetchUserTable,
    deleteUser,
} from '../../actions/index.js'
import VMTable from './components/VMTable.js'
import UserTable from './components/UserTable.js'
import CustomerTable from './components/CustomerTable.js'
import DiskTable from './components/DiskTable.js'

class Dashboard extends Component {
    constructor(props) {
        super(props)
        this.state = {
            showPopup: false,
            loaded: false,
            userTableFetched: false,
        }
    }

    componentDidMount() {
        this.props.dispatch(fetchUserActivity(false))
    }

    componentDidUpdate(prevProps) {
        if (
            this.props.access_token &&
            this.props.userTable.length === 0 &&
            !this.state.userTableFetched
        ) {
            this.props.dispatch(fetchUserTable())
            this.setState({ userTableFetched: true })
        }
    }

    deleteUser = (user) => {
        this.props.dispatch(deleteUser(user))
    }

    render() {
        return (
            <div>
                <div
                    style={{
                        marginTop: 5,
                        fontSize: 45,
                        fontWeight: 'bold',
                        marginBottom: 60,
                        width: 275,
                        color: '#111111',
                    }}
                >
                    DASHBOARD
                </div>
                <div style={{ display: 'flex', marginTop: 5 }}>
                    <div
                        style={{
                            display: 'block',
                            width: '100%',
                            position: 'relative',
                            bottom: 36,
                        }}
                    >
                        <Tabs>
                            <TabList
                                style={{
                                    textAlign: 'left',
                                    border: 'none',
                                    fontSize: 16,
                                }}
                            >
                                <Tab>Cloud PCs</Tab>
                                <Tab>Disks</Tab>
                                <Tab>Users</Tab>
                                <Tab>Customers</Tab>
                            </TabList>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <VMTable />
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <DiskTable />
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <UserTable />
                            </TabPanel>
                            <TabPanel style={{ paddingTop: 20 }}>
                                <CustomerTable />
                            </TabPanel>
                        </Tabs>
                    </div>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        vm_info: state.AccountReducer.vm_info
            ? state.AccountReducer.vm_info
            : [],
        vmsUpdated: state.AccountReducer.vmsUpdated,
        activityFetched: state.AccountReducer.activityFetched,
        userActivity: state.AccountReducer.userActivity
            ? state.AccountReducer.userActivity
            : [],
        userTable: state.AccountReducer.userTable
            ? state.AccountReducer.userTable
            : [],
        usersUpdated: state.AccountReducer.usersUpdated,
        access_token: state.AccountReducer.access_token,
        login_attempts: state.AccountReducer.login_attempts,
    }
}

export default connect(mapStateToProps)(Dashboard)
