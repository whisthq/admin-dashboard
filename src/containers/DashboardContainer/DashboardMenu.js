import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Dropdown } from 'react-bootstrap'
import {
    faHome,
    faChartArea,
    faList,
    faExternalLinkAlt,
} from '@fortawesome/free-solid-svg-icons'
import { FaLink } from 'react-icons/fa'

import 'static/App.css'

import Logo from 'assets/logo.svg'

withRouter((props) => <DashboardMenu {...props} />)

class DashboardMenu extends Component {
    constructor(props) {
        super(props)
        this.state = { page: 'dashboard' }
    }

    render() {
        return (
            <div
                style={{
                    backgroundColor: 'white',
                    height: '100%',
                    width: 75,
                    minHeight: '100vh',
                    paddingRight: 25,
                    paddingLeft: 0,
                    paddingTop: 50,
                }}
            >
                <div style={{ paddingLeft: 25 }}>
                    <img
                        src={Logo}
                        alt=""
                        style={{ width: 30, height: 30, marginBottom: 70 }}
                    />
                </div>
                <div style={{ marginTop: 40, fontSize: 18 }}>
                    <div style={{ marginBottom: 30, marginLeft: 5 }}>
                        <Link
                            style={{
                                textDecoration: 'none',
                                color:
                                    this.props.page === 'dashboard'
                                        ? '#111111'
                                        : '#999999',
                                fontWeight:
                                    this.props.page === 'dashboard'
                                        ? 'bold'
                                        : 'normal',
                                borderLeft:
                                    this.props.page === 'dashboard'
                                        ? 'solid 4px #6241e8'
                                        : 'solid 4px white',
                                paddingLeft: 18,
                            }}
                            to="/admin/dashboard"
                        >
                            <FontAwesomeIcon
                                icon={faHome}
                                style={{
                                    color:
                                        this.props.page === 'dashboard'
                                            ? '#111111'
                                            : '#d0ced9',
                                    fontSize: 20,
                                }}
                            />
                        </Link>
                    </div>
                    <div style={{ marginBottom: 30, marginLeft: 5 }}>
                        <Link
                            style={{
                                textDecoration: 'none',
                                fontWeight:
                                    this.props.page === 'analytics'
                                        ? 'bold'
                                        : 'normal',
                                borderLeft:
                                    this.props.page === 'analytics'
                                        ? 'solid 4px #6241e8'
                                        : 'solid 4px white',
                                paddingLeft: 20,
                            }}
                            to="/admin/analytics"
                        >
                            <FontAwesomeIcon
                                icon={faChartArea}
                                style={{
                                    color:
                                        this.props.page === 'analytics'
                                            ? '#111111'
                                            : '#d0ced9',
                                    fontSize: 20,
                                }}
                            />
                        </Link>
                    </div>
                    <div style={{ marginBottom: 30, marginLeft: 5 }}>
                        <Link
                            style={{
                                textDecoration: 'none',
                                fontWeight:
                                    this.props.page === 'logs'
                                        ? 'bold'
                                        : 'normal',
                                borderLeft:
                                    this.props.page === 'logs'
                                        ? 'solid 4px #6241e8'
                                        : 'solid 4px white',
                                paddingLeft: 21,
                            }}
                            to="/admin/logs"
                        >
                            <FontAwesomeIcon
                                icon={faList}
                                style={{
                                    color:
                                        this.props.page === 'logs'
                                            ? '#111111'
                                            : '#d0ced9',
                                    fontSize: 19,
                                }}
                            />
                        </Link>
                    </div>
                    <Dropdown
                        style={{
                            textDecoration: 'none',
                            paddingLeft: 29,
                        }}
                    >
                        <Dropdown.Toggle variant="success" id="dropdown-basic">
                            <FontAwesomeIcon
                                icon={faExternalLinkAlt}
                                style={{
                                    color: '#d0ced9',
                                    fontSize: 18,
                                    outline: 'none',
                                }}
                            />
                        </Dropdown.Toggle>

                        <Dropdown.Menu style={{ border: 'none', padding: 20 }}>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://s3.console.aws.amazon.com/s3/home?region=us-east-1#"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    AWS S3
                                </a>
                            </Dropdown.Item>{' '}
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://lightsail.aws.amazon.com/ls/webapp/home/instances?#"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    AWS Lightsail
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://dashboard.heroku.com/teams/fractalcomputers/apps"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Heroku
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://fractalcomputers.postman.co/workspaces/6ded464f-396a-4792-a5c9-5ac7c689d269/apis"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Postman
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://drive.google.com/open?id=1kbpGbVeYd5LVJgzGN0dKro-ryizQr2kg"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Google Drive
                                </a>
                            </Dropdown.Item>
                            <Dropdown.Item style={{ marginBottom: 5 }}>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://www.notion.so/fractalcomputers/2fa59d87701b4ec7967377450c75008a?v=f20e6d7d6d5f4d4c9a0f886871259833"
                                >
                                    <FaLink
                                        style={{
                                            marginRight: 10,
                                            color: '#d0ced9',
                                            fontSize: 12,
                                        }}
                                    />
                                    Notion
                                </a>
                            </Dropdown.Item>
                        </Dropdown.Menu>
                    </Dropdown>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        page: state.AccountReducer.page,
    }
}

export default connect(mapStateToProps)(DashboardMenu)
