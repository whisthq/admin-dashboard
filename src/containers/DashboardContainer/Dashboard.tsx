import React from 'react'
import { connect } from 'react-redux'
import moment from 'moment'
import { Route, Switch, Redirect } from 'react-router-dom'
import { Button } from 'react-bootstrap'

import { fetchUserActivity, logout } from '../../actions/index'

import DashboardMenu from './DashboardMenu'
import Database from './DatabasePage/Database'
import Logs from './LogsPage/Logs'
import Analytics from './AnalyticsPage/Analytics'

import '../../static/App.css'
import 'react-tabs/style/react-tabs.css'


class Admin extends React.Component<any, any> {
    constructor(props: any) {
        super(props)
        this.state = {
            date: 0,
        }
        this.updateWindowDimensions = this.updateWindowDimensions.bind(this)
    }

    componentDidMount() {
        this.setState({
            date: moment(Date.now()).format('MMMM Do, YYYY'),
        })
        this.props.dispatch(fetchUserActivity())
    }

    updateWindowDimensions() {
        this.setState({ width: window.innerWidth, height: window.innerHeight })
    }

    render() {
        return (
            <div style={{ backgroundColor: '#f3f2f5', minHeight: '100vh' }}>
                {this.props.authenticated ? (
                    <div style={{ display: 'flex' }}>
                        <div style={{ width: 75, margin: 0 }}>
                            <DashboardMenu />
                        </div>
                        <div
                            style={{
                                width: 'calc(100%-75px)',
                                margin: 0,
                                overflowX: 'hidden',
                            }}
                        >
                            <div className="p-5">
                                <div className="d-flex justify-content-between">
                                    <div>{this.state.date}</div>
                                    <Button
                                        onClick={() =>
                                            this.props.dispatch(logout())
                                        }
                                        style={{
                                            border: 'none',
                                            padding: '10px 30px',
                                            fontWeight: 'bold',
                                            color: '#1ba8e0',
                                            background:
                                                'rgba(94, 195, 235, 0.2)',
                                            borderRadius: 3,
                                        }}
                                    >
                                        Logout
                                    </Button>
                                </div>
                                <Switch>
                                    <Route
                                        exact
                                        path="/admin"
                                        component={Database}
                                    />
                                    <Route
                                        path="/admin/dashboard"
                                        component={Database}
                                    />
                                    <Route
                                        path="/admin/logs"
                                        component={Logs}
                                    />
                                    <Route
                                        path="/admin/analytics"
                                        component={Analytics}
                                    />
                                </Switch>
                            </div>
                        </div>
                    </div>
                ) : (
                    <Redirect to="/" />
                )}
            </div>
        )
    }
}

function mapStateToProps(state: any) {
    return {
        authenticated: state.AccountReducer.authenticated,
    }
}

export default connect(mapStateToProps)(Admin)
