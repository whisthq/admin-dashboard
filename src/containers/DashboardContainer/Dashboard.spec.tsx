import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { Button } from 'react-bootstrap'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { Admin } from './Dashboard'
import { DashboardMenu } from './DashboardMenu'

Enzyme.configure({ adapter: new Adapter() })

/*
The admin should render a switch to the four components it can display and a <DashboardMenu />.
It should also have a logout button.
 */
describe('<Admin />', () => {
  test('check that componentDidMount fetches user activity', () => {
      //TODO
  })
  test('check that componentDidMount sets the date', () => {
      // TODO
  })
  test('updateWindowdimensions() sets state based on window.*', () => {
    // TODO
  })

  it('renders a DashboardMenu', () => {
    const wrapper = shallow(<Admin />)
    expect(wrapper.find(DashboardMenu)).toHaveLength(1)
  })
  
  it('renders one <Switch /> components', () => {
    const wrapper = shallow(<Admin />)
    expect(wrapper.find(Switch)).toHaveLength(1)
  }) 

  it('renders four <Route /> components', () => {
      const wrapper = shallow(<Admin />)
      expect(wrapper.find(Route)).toHaveLength(4)
  })

  it('renders the logout button', () => {
      const wrapper = shallow(<Admin />)
      expect(wrapper.find(Button)).toHaveLength(1)
  })
})