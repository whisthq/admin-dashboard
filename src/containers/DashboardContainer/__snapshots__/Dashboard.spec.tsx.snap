// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f3f2f5",
      "minHeight": "100vh",
    }
  }
>
  <Redirect
    to="/"
  />
</div>
`;
